### PR TITLE
Add a getting started guide for the Raspberry Pi

### PR DESCRIPTION
**Pull request by rnestler**

### CODE REVIEW

File: docs/getting-started-on-the-raspberry-pi.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+This guide assumes you are using the [Raspbian image]. It should work on other
+distributions as well, but some details may be different.
+
+We will compile the driver on the Raspberry Pi itself, of course you can also
+cross compile from your system. 
+
+## Download Driver
+
+Get the latest release from
+https://github.com/Sensirion/embedded-uart-sps/releases (sps30-uart-x.y.z.zip)
```
Comment (rnestler) [inline docs/getting-started-on-the-raspberry-pi.md]: I used 1.0.0 for the rest of the guide. I'm not sure if we should use real version numbers or just `x.y.z` everywhere.

File: docs/getting-started-on-the-raspberry-pi.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
+# Getting Started on the Raspberry Pi 3
+
+This guide assumes you are using the [Raspbian image]. It should work on other
+distributions as well, but some details may be different.
+
+We will compile the driver on the Raspberry Pi itself, of course you can also
+cross compile from your system.
```
Comment (rnestler) [inline docs/getting-started-on-the-raspberry-pi.md]: Should I add a note about ssh and internet access on the Pi?

File: docs/getting-started-on-the-raspberry-pi.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+```bash
+# on the Raspberry Pi
+$ wget https://github.com/Sensirion/embedded-uart-sps/releases/download/1.0.0/sps30-uart-1.0.0.zip
+$ unzip sps30-uart-1.0.0.zip
+```
+
+## Get Port of the Connected Sensor
+
+We need to now the port of the connected sensor. After connecting the sensor
```
Comment (abrauchli) [inline docs/getting-started-on-the-raspberry-pi.md]: now -> know

File: docs/getting-started-on-the-raspberry-pi.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+via USB run the following:
+```bash
+# on the Raspberry Pi
+$ dmesg | grep usb
```
Comment (abrauchli) [inline docs/getting-started-on-the-raspberry-pi.md]: Isn't `ls /dev/ttyUSB*` simpler? I think there's no other USB tty by default on a raspi..

File: docs/getting-started-on-the-raspberry-pi.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+cross compile from your system. 
+
+## Download Driver
+
+Get the latest release from
+https://github.com/Sensirion/embedded-uart-sps/releases (sps30-uart-x.y.z.zip)
+
+```bash
+# on the Raspberry Pi
+$ wget https://github.com/Sensirion/embedded-uart-sps/releases/download/1.0.0/sps30-uart-1.0.0.zip
```
Comment (abrauchli) [inline docs/getting-started-on-the-raspberry-pi.md]: I think we should explain that `$` is the user prompt and not part of the command (there's also output later in the file)..

File: docs/getting-started-on-the-raspberry-pi.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+Then in `sensirion_uart_implementation.c` change the line with `#define TTYDEV`
+to match the device you got from the step before.
+```c
+#define TTYDEV "/dev/ttyUSB0"
+```
+
+## Compile and Run
+
+Now we are ready to compile and run the sample:
```
Comment (abrauchli) [inline docs/getting-started-on-the-raspberry-pi.md]: Add a note that every time the device is unplugged, the devicepath might change and the binary must be recompiled.

File: docs/getting-started-on-the-raspberry-pi.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
+# Getting Started on the Raspberry Pi 3
```
Comment (abrauchli) [inline docs/getting-started-on-the-raspberry-pi.md]: Cool, I think we should also link this file from the main README

File: docs/getting-started-on-the-raspberry-pi.md (hunk 2 above)
Comment (winkj) [reply docs/getting-started-on-the-raspberry-pi.md]: I think it's fair to assume that the user set that up before; also see **https://github.com/winkj/raspberrypi-ess#introduction**

File: docs/getting-started-on-the-raspberry-pi.md (hunk 1 above)
Comment (winkj) [reply docs/getting-started-on-the-raspberry-pi.md]: maybe link to https://github.com/Sensirion/embedded-uart-sps/releases/latest instead?

I think using either the release number or x.y.z is fine

File: docs/getting-started-on-the-raspberry-pi.md (hunk 4 above)
Comment (rnestler) [reply docs/getting-started-on-the-raspberry-pi.md]: Is it guaranteed that it shows up as `/dev/ttyUSB`? I have also seen stuff like `/dev/ttyACM0`.

File: docs/getting-started-on-the-raspberry-pi.md (hunk 7 above)
Comment (rnestler) [reply docs/getting-started-on-the-raspberry-pi.md]: Already done.

File: docs/getting-started-on-the-raspberry-pi.md (hunk 4 above)
Comment (abrauchli) [reply docs/getting-started-on-the-raspberry-pi.md]: with both official cables yes, IIRC, ACM is used if the device also supports modem-like AT commands.

File: docs/getting-started-on-the-raspberry-pi.md (hunk 1 above)
Comment (rnestler) [reply docs/getting-started-on-the-raspberry-pi.md]: Fixed.

File: docs/getting-started-on-the-raspberry-pi.md (hunk 3 above)
Comment (rnestler) [reply docs/getting-started-on-the-raspberry-pi.md]: Fixed.

File: docs/getting-started-on-the-raspberry-pi.md (hunk 6 above)
Comment (rnestler) [reply docs/getting-started-on-the-raspberry-pi.md]: Didn't happen during my testing, but true that *may* happen.

Comment (rnestler) [reply docs/getting-started-on-the-raspberry-pi.md]: I added a trouble shooting section which explains this among other errors.

File: docs/getting-started-on-the-raspberry-pi.md (hunk 5 above)
Comment (rnestler) [reply docs/getting-started-on-the-raspberry-pi.md]: Done.

File: docs/getting-started-on-the-raspberry-pi.md (hunk 2 above)
Comment (rnestler) [reply docs/getting-started-on-the-raspberry-pi.md]: Thanks. I changed it accordingly.

File: docs/getting-started-on-the-raspberry-pi.md (hunk 4 above)
Comment (rnestler) [reply docs/getting-started-on-the-raspberry-pi.md]: The dmesg has the advantage that it is also a good way for debugging any issues with the USB device, so I'd rather keep it.

Comment (winkj) [reply docs/getting-started-on-the-raspberry-pi.md]: I'd say we can keep this for now, and improve it later on if it is confusing to users

File: docs/getting-started-on-the-raspberry-pi.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+
+This guide assumes you are using the [Raspbian image]. It should work on other
+Linux based setups as well, but some details may be different.
+
+We will compile the driver on the Raspberry Pi itself so we assume that the
+reader has basic knowledge of the Raspberry Pi, shell access to it and that is
```
Comment (abrauchli) [inline docs/getting-started-on-the-raspberry-pi.md]: typo in last word `is` -> `it`

Comment (rnestler) [reply docs/getting-started-on-the-raspberry-pi.md]: Duh. Fixed.

**Pull request merged**